programs/compressed-token/src/process_transfer.rs

```diff
@@ -243,19 +252,20 @@ pub struct TransferInstruction<'info> {
     pub self_program: Program<'info, crate::program::PspCompressedToken>,
 }
 
-// TODO: parse utxos a more efficient way, since owner is sent multiple times this way
+// TODO: parse compressed_accounts a more efficient way, since owner is sent multiple times this way
 // This struct is equivalent to the InstructionDataTransfer, but uses the imported types from the psp_compressed_pda
 #[derive(Debug, Clone, AnchorSerialize, AnchorDeserialize)]
 pub struct InstructionDataTransfer {
     proof: Option<CompressedProof>,
     root_indices: Vec<u16>,
-    in_utxos: Vec<InUtxoTuple>,
-    in_tlv_data: Vec<TokenTlvData>,
-    out_utxos: Vec<TokenTransferOutUtxo>,
+    input_compressed_accounts_with_merkle_context: Vec<CompressedAccountWithMerkleContext>,
+    input_token_data: Vec<TokenData>,
+    output_compressed_accounts: Vec<TokenTransferOutputData>,
+    output_state_merkle_tree_account_indices: Vec<u8>,
 }
 
 #[derive(Clone, Copy, Debug, PartialEq, Eq, AnchorSerialize, AnchorDeserialize)]
-pub struct TokenTransferOutUtxo {
+pub struct TokenTransferOutputData {
```

Comment:
`TokenTransferOutputData` still has `index_mt_account` so we're currently sending the same data to the chain twice. once as part of the `TokenTransferOutputData`  and once in the  `output_state_merkle_tree_account_indices` of the InstructionDataTransfer of compressed-token in line 487

Reply:
fixed